### ISSUE TITLE
Typed JSON: editable field clash with method name

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#3089]
Having this model

**Foo.json**
```json
{
    "Name$": "",
    "AddNew$" : 0
}
```

with this code-behind

```C#
partial class Foo : Json {
  void Handle(Input.AddNew a) {
    AddNew();
  }

  void AddNew() {}
}
```

prevents any Starcounter application from compiling:

Code | Description
--- | ---
Error CS0102 | The type 'Foo' already contains a definition for 'AddNew'

This is probably known, and the reason for it too: because the `AddNew$` generates an integer named "AddName" in the generated class, and that field clash with my method.

Why I want to bring it up still is because I want to see if we can get around it somehow. I think that pattern will be quite common - having a "button" with a name like "Save" and then a "Save()" method in the code-behind.

**Comment by chrhol** [comment]
Yeah, one problem with the pattern described above is that there is no specific type for triggers/actions in JSON, which means that all properties in JSON-by-example will have a value, even if the value is only used to trigger the callback. 

We had a workaround before, by setting a value to `null` in JSON-by-example and use that as a trigger. Then we could skip generate a property for those and the sample above would work. However we decided that we wanted to avoid workarounds and special handling so we started removing that part.

I think this might be difficult to solve since the .Net c# compiler does not allow a property (or field) and a method to have the same name in a class. So it's not really something we can affect, if we want to keep the original thought avout how TypedJson should work. 

I guess we could add some way to say that a specific property in JSON-by-example should not generate a property in generated code, but then again it feels easier just to change name of the `AddNew` method instead.

**Comment by warpech** [comment]
It was also reported as a quirk by @jwillmer, but when I explained it (similarly to what @chrhol explains above), it got clearer.

@chrhol any idea how to improve the docs on this?

**Comment by per-samuelsson** [comment]
> @chrhol any idea how to improve the docs on this?

I don't. Yet. But I've been obsessed trying to get deeper into the whole JSON-domain during the weekend, and come up with ways to write improved (and more) documentation, on several areas. Let's hope this can result in anything, if @chrhol doesnt come up with a super-pattern! :smile:
